lib/pdf_fill/processors/base_form_4142_processor.rb

```diff
@@ -0,0 +1,175 @@
+require 'pdf_fill/forms/va218940'
+require 'pdf_fill/forms/va1010cg'
+require 'pdf_utilities/datestamp_pdf'
+require 'simple_forms_api_submission/metadata_validator'
+
+module Processors
+  class Form4142ValidationError < StandardError
+    def initialize(errors)
+      super
+      @errors = errors
+    end
+  end
+
+  class BaseForm4142Processor
+    SIGNATURE_DATE_KEY = 'signatureDate'
+    SIGNATURE_TIMESTAMP_FORMAT = '%Y-%m-%d %H:%M:%S'
+    TIMEZONE = 'Central Time (US & Canada)'
+    FORM_SCHEMA_ID = '21-4142'
+    LEGACY_FORM_CLASS_ID = '21-4142'
+    FORM_CLASS_ID_2024 = '21-4142-2024'
+
+    # @return [Pathname] the generated PDF path
+    # @return [Hash] the generated request body
+    attr_reader :pdf_path, :request_body
+
+    def initialize(validate: true)
+      validate_form4142 if validate
+      @pdf_path = generate_stamp_pdf
+      @request_body = {
+        'document' => to_faraday_upload,
+        'metadata' => generate_metadata
+      }
+    end
+
+    # Single entry point: choose the right form_class_id, then fill & stamp.
+    def generate_stamp_pdf
+      selected_form_class_id = generate_2024_version? ? FORM_CLASS_ID_2024 : LEGACY_FORM_CLASS_ID
+
+      pdf = PdfFill::Filler.fill_ancillary_form(
+        form_data,
+        pdf_identifier,
+        selected_form_class_id
+      )
+
+      # Handle signature stamping for 2024 version
+      if signature.present? && selected_form_class_id == FORM_CLASS_ID_2024
+        pdf = PDFUtilities::DatestampPdf.new(pdf).run(
+          text: signature,
+          x: 50,
+          y: 560,
+          text_only: true,
+          size: 10,
+          page_number: 1,
+          template: pdf,
+          multistamp: true,
+          timestamp_required: false
+        )
+      end
+
+      # Add VA.gov timestamp
+      stamped_path = PDFUtilities::DatestampPdf.new(pdf).run(
+        text: 'VA.gov',
+        x: 5,
+        y: 5,
+        timestamp: submission_date
+      )
+
+      # Add VA.gov Submission text
+      stamped_path = PDFUtilities::DatestampPdf.new(stamped_path).run(
+        text: 'VA.gov Submission',
+        x: 510,
+        y: 775,
+        text_only: true
+      )
+    end
+
+    protected
+
+    # Default: use legacy form
+    def generate_2024_version?
+      false
+    end
+    # Abstract methods to be implemented by subclasses
+    def form_data
+      raise NotImplementedError, 'Subclasses must implement form_data method'
+    end
+
+    def pdf_identifier
+      raise NotImplementedError, 'Subclasses must implement pdf_identifier method'
+    end
+
+    def metadata_uuid
+      raise NotImplementedError, 'Subclasses must implement metadata_uuid method'
+    end
+
+    def submission_date
+      raise NotImplementedError, 'Subclasses must implement submission_date method'
+    end
+
+    def country_code_for_us_validation
+      'US' # Default implementation, can be overridden
+    end
+
+    private
+
+    def uuid
+      @uuid ||= SecureRandom.uuid
```

Comment:
If we are the only ones that use the uuid (and it's not being used in the base processor), we should probably just move it into our processor.